src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -417,12 +417,29 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             vkey != VK_SNAPSHOT &&
             keyDown)
         {
-            if (_terminal->IsInMarkMode() && modifiers.IsCtrlPressed() && vkey == 'A')
+            if (_terminal->IsInMarkMode())
             {
-                auto lock = _terminal->LockForWriting();
-                _terminal->SelectAll();
-                _updateSelection();
-                return true;
+                if (modifiers.IsCtrlPressed() && vkey == 'A')
+                {
+                    auto lock = _terminal->LockForWriting();
+                    _terminal->SelectAll();
+                    _updateSelection();
+                    return true;
+                }
+                else if (_settings->DetectURLs() && vkey == VK_TAB)

```

Comment:
If you check the `vkey` first the compiler will skip the slower COM call (boolean short circuit evaluation).